### DISCUSSION TITLE
get players position at x roundtime

### DISCUSSION THREAD

**Discussion by RaphaelHippe** [saul/demofile#223] [Questions] [answered]
Hey, me again ;-), 

first of all: I saw this issue: https://github.com/saul/demofile/issues/3 and tried to use it. 
but I don't want to get the position of each player every tick or every event but at a set round timer mark so e.g. after 30 seconds have past in each round. 

```javascript
    demoFile.on('tickend', e => {
      let currentRoundTime = demoFile.currentTime - currentRoundStarttime;
      if (currentRoundTime > 30.0 && currentRoundTime < 30.025) {
        console.log(e, demoFile.currentTime, currentRoundTime);
        console.log('NEW ROUND 30 SEC MARK');
        for (var i = 0; i < myUserIds.length; i++) {
          let ent = demoFile.entities.entities[myUserIds[i]];
          console.log(myUserIds[i], ent.position);
          let xy = ent.getProp('DT_CSNonLocalPlayerExclusive', 'm_vecOrigin');
          console.log('xy', xy);
          let pos = {
            x: xy.x,
            y: xy.y,
            z: ent.getProp('DT_CSNonLocalPlayerExclusive', 'm_vecOrigin[2]')
          }
          console.log(myUserIds[i], pos);
        }
      }
    });


demoFile.gameEvents.on('round_start', e => {
      currentRoundStarttime = demoFile.currentTime;
    });
```

The issues I have with this are: 
- multiple results for the same second because ticks happen faster than seconds
- different amount of results per round (sometimes 2, sometimes 3 and so on) 
- `xy` seems to be undefined sometimes: 

```
x: xy.x,
TypeError: Cannot read property 'x' of undefined
```

after that I tried the `ent.position` but that returns the same position everytime (every round, and every tick) for each player, which I find very odd. 

Thank, 

Raphael Hippe

**Comment by saul** [comment, marked as answer]
Hi Raphael,

A couple of points to note:

* Entity index are not the same as user IDs. Entities represent all networked objects in the game (this can include props, doors, etc), so you're most likely not getting the position of the players in this instance.
* You should check whether you're at a specific tick in `tickend` instead of doing a range check for currentTime. It's more efficient and more accurate.
* You can just use `ent.position` on players instead of calculating it manually - you're looking at an old comment. I've edited the comment that you've linked to.

With that said, I've fixed your code here - let me know if it works:

```js
let checkpointTick = -1;

demoFile.on('tickend', e => {
  if (demoFile.currentTick == checkpointTick) {
    console.log(e, demoFile.currentTime, demoFile.currentTick);
    console.log('NEW ROUND 30 SEC MARK');

    for (var i = 0; i < myUserIds.length; i++) {
      let ent = demoFile.entities.getByUserId(myUserIds[i]);
      console.log(myUserIds[i], ent.position);
    }
  }
});

demoFile.gameEvents.on('round_start', e => {
  let ticksPerSec = demoFile.header.playbackTicks / demoFile.header.playbackTime;

  // Calculate the tick that is 30 seconds in the future.
  // We do `| 0` to make the number whole (we only ever deal with ticks that are integers).
  checkpointTick = (demoFile.currentTick + (30 * ticksPerSec)) | 0;
});
```

Thanks

**Comment by RaphaelHippe** [comment]
This seems to work. 
Loved the fast and helpful answer(s)!

Thank you very much :-)
Cheers, 
Raphael
